UI/MapWnd.cpp

```diff
@@ -283,6 +284,198 @@ namespace {
+            GG::Pt part_lr(LR.x - m_margin, LR.y - m_margin);
+            GG::BeveledRoundedRectangle(part_ul, part_lr, BG_CLR, BORDER_CLR, false);
+        }
+
+        void DoLayout() {
+            const GG::Y row_height{ClientUI::Pts()};
+            const GG::Y offset{32};
+            const GG::X label_width{m_col_widths.at(0) - (m_margin * 2)};
+            const GG::X value_width{m_col_widths.at(1) - (m_margin * 3)};
+
+            GG::Pt label_ul{GG::X{m_margin}, offset + m_margin};
+            GG::Pt label_lr{label_ul.x + label_width, offset + row_height};
+            GG::Pt value_ul{label_lr.x + m_margin, label_ul.y};
+            GG::Pt value_lr{value_ul.x + value_width, label_lr.y};
+
+            const GG::Pt next_row{GG::X0, row_height + (m_margin * 2)};
+            const GG::Pt space_row{next_row * 5 / 4};
+
+            LayoutRow("MAP_FLEET_SHIP_COUNT",       label_ul, label_lr, value_ul, value_lr, space_row);
+            LayoutRow("MAP_FLEET_ARMED_COUNT",      label_ul, label_lr, value_ul, value_lr, next_row);
+            LayoutRow("MAP_FLEET_UNARMED_COUNT",    label_ul, label_lr, value_ul, value_lr, space_row);
+            LayoutRow("MAP_FLEET_CARRIER_COUNT",    label_ul, label_lr, value_ul, value_lr, next_row);
+            LayoutRow("MAP_FLEET_TROOP_COUNT",      label_ul, label_lr, value_ul, value_lr, next_row);
+            LayoutRow("MAP_FLEET_COLONY_COUNT",     label_ul, label_lr, value_ul, value_lr, space_row);
+            LayoutRow("MAP_FLEET_SLOT_COUNT",       label_ul, label_lr, value_ul, value_lr, next_row);
+            LayoutRow("MAP_FLEET_PART_COUNT",       label_ul, label_lr, value_ul, value_lr, GG::Pt(GG::X0, GG::Y0));
+
+            Resize(GG::Pt(value_lr.x + (m_margin * 3), value_lr.y + (m_margin * 3)));
+        }
+
+        /** Constructs and attaches new label and value controls for the given label. */
+        void NewLabelValue(const std::string& label, bool title = false) {
+            if (m_labels.find(label) != m_labels.end())
+                return;
+
+            GG::Y height{ClientUI::Pts()};
+            // center format for title label
+            m_labels.emplace(label, std::make_pair(
+                std::shared_ptr<CUILabel>{
+                    new CUILabel{UserString(label),
+                        title ? GG::FORMAT_CENTER : GG::FORMAT_RIGHT,
+                        GG::NO_WND_FLAGS, GG::X0, GG::Y0,
+                        m_col_widths.at(0) - (m_margin * 2), height
+                    }
+                },
+                std::shared_ptr<CUILabel>{
+                    new CUILabel{"0", GG::FORMAT_RIGHT,
+                        GG::NO_WND_FLAGS, GG::X0, GG::Y0,
+                        m_col_widths.at(1) - (m_margin * 2), height
+                    }
+                }
+            ));
+
+            if (title) {  // utilize bold font for title label and value
+                m_labels.at(label).first.get()->SetFont(ClientUI::GetBoldFont());
```

Comment:
are these `.get()` calls necessary? You should be able to dereference a shared_ptr directly...